src/polynomial.h

```diff
@@ -0,0 +1,67 @@
+#ifndef CSYMPY_POLYNOMIALS_H
+#define CSYMPY_POLYNOMIALS_H
+
+#include "monomials.h"
+#include "dict.h"
+#include "basic.h"
+#include "integer.h"
+
+namespace CSymPy {
+
+    class Polynomial : public Basic{
+    public:
+        int degree;
+        std::string var_;
+        //umap_vec_mpz dict;
+        map_integer_uint dict_;
```

Comment:
Currently it is the other way round (exponents in `integer`). I used `map_integer_uint` just to test the code. I'll update it to use `uint` for exponent and `rational` (or `integer`) for coefficients.

Ultimately, however coefficients should depend on the base ring it is being defined on.